mcp/src/minecraft/com/elementfx/tvp/ad/inventory/SlotGoblinCrafting.java

```diff
@@ -0,0 +1,63 @@
+package com.elementfx.tvp.ad.inventory;
+
+import javax.annotation.Nullable;
+
+import com.elementfx.tvp.ad.item.crafting.ElvenCraftingManager;
+import com.elementfx.tvp.ad.item.crafting.GoblinCraftingManager;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.inventory.SlotCrafting;
+import net.minecraft.item.ItemStack;
+
+public class SlotGoblinCrafting extends SlotCrafting
+{
+	/** The craft matrix inventory linked to this result slot. */
+    private final InventoryCrafting craftMatrix;
+
+    /** The player that is using the GUI where this slot resides. */
+    private final EntityPlayer thePlayer;
+
+	public SlotGoblinCrafting(EntityPlayer player, InventoryCrafting craftingInventory, IInventory inventoryIn, int slotIndex, int xPosition, int yPosition) 
+	{
+		super(player, craftingInventory, inventoryIn, slotIndex, xPosition, yPosition);
+		this.thePlayer = player;
+        this.craftMatrix = craftingInventory;
+	}
+	
+	public void onPickupFromSlot(EntityPlayer playerIn, ItemStack stack)
+    {
+        this.onCrafting(stack);
+        ItemStack[] aitemstack = GoblinCraftingManager.getInstance().getRemainingItems(this.craftMatrix, playerIn.worldObj);
```

Comment:
Is this the only reason we need custom slots? Can you think of any other reason we would need custom slots in the future?